### PR TITLE
feat(theme-panel): 新增主题风格配置组件

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

✨ Feature

- [x] New feature

### 📝 Description

```tsx
 <ThemePanel
    title="主题配置"
    disableCustomPrimaryColorPicker={false}
    defaultCollapsed={false}
    onChange={(options, theme) => {
      console.log('onChange:', options, theme);
    }}
    onReset={(options, prevOptions, theme) => {
      console.log('onReset:', options, prevOptions, theme);
    }}
  />
```

### 🖼️ Screenshot

Kapture 2024-06-12 at 16 44 10

image

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-react-components/src/components/common/radio-group/index.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { S2_PREFIX_CLS } from '@antv/s2';
```
Comment (lijinke666) [inline packages/s2-react-components/src/components/common/radio-group/index.tsx:1]: image

File: packages/s2-react-components/src/components/common/reset-button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ReloadOutlined } from '@ant-design/icons';
```
Comment (lijinke666) [inline packages/s2-react-components/src/components/common/reset-button/index.tsx:1]: image

File: packages/s2-react-components/src/components/common/reset-group/index.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { CaretRightOutlined } from '@ant-design/icons';
```
Comment (lijinke666) [inline packages/s2-react-components/src/components/common/reset-group/index.tsx:1]: image

File: packages/s2-react-components/src/components/config/theme-panel/color-box/index.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { S2_PREFIX_CLS } from '@antv/s2';
```
Comment (lijinke666) [inline packages/s2-react-components/src/components/config/theme-panel/color-box/index.tsx:1]: image

File: packages/s2-react-components/src/components/config/theme-panel/color-picker-panel/index.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
+import { CaretRightOutlined } from '@ant-design/icons';
+import { S2_PREFIX_CLS, i18n } from '@antv/s2';
+import { Collapse } from 'antd';
```
Comment (lijinke666) [inline packages/s2-react-components/src/components/config/theme-panel/color-picker-panel/index.tsx:3]: image

File: packages/s2-react-components/src/components/config/theme-panel/index.tsx [hunk 6]
```diff
@@ -1,5 +1,227 @@
+import { S2_PREFIX_CLS, i18n } from '@antv/s2';
```
Comment (lijinke666) [inline packages/s2-react-components/src/components/config/theme-panel/index.tsx:1]: image

File: packages/s2-core/src/utils/theme.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ import { injectCssText } from './inject-css-text';
 export const getPalette = (type?: ThemeName): Palette => {
   const themeName = type || 'default';
```
Comment (wjgogogo) [inline packages/s2-core/src/utils/theme.ts:11]: 那上面这句是不是可以干掉了

File: packages/s2-react-components/src/components/common/reset-group/index.tsx (hunk 3 above)
Comment (wjgogogo) [reply packages/s2-react-components/src/components/common/reset-group/index.tsx:1]: 收起时，宽度会变窄？

File: packages/s2-react-components/src/components/config/theme-panel/index.tsx [hunk 8]
```diff
@@ -1,5 +1,227 @@
...
+        [field]: e.target.value,
+      };
+
+      setOptions(newOptions);
+    };
+
+  const renderIcon = (label: React.ReactNode, Component: React.ReactNode) => {
+    return (
+      <Tooltip title={label}>
+        <>{Component}</>
```
Comment (wjgogogo) [inline packages/s2-react-components/src/components/config/theme-panel/index.tsx:114]: 外层的 <></> 没必要吧？

File: packages/s2-react-components/src/components/common/reset-group/index.tsx (hunk 3 above)
Comment (lijinke666) [reply packages/s2-react-components/src/components/common/reset-group/index.tsx:1]: fixed.

File: packages/s2-core/src/utils/theme.ts (hunk 7 above)
Comment (lijinke666) [reply packages/s2-core/src/utils/theme.ts:11]: resolved

File: packages/s2-react-components/src/components/config/theme-panel/index.tsx (hunk 8 above)
Comment (lijinke666) [reply packages/s2-react-components/src/components/config/theme-panel/index.tsx:114]: fixed.

**Review by wjgogogo — APPROVED** [review]

**Pull request merged**